### ISSUE TITLE
Using an observer pattern?

### ISSUE THREAD

**Issue by manno-xx** [watsonfilipp/DevelopingForMobileFilipp#7]
I'd look into the observer pattern here probably. (https://gamedevbeginner.com/events-and-delegates-in-unity/)

The shooting and updating the UI should be two separate things. The UI however depends on the shooting script. People could say that it is a 'view' on the shooting, displaying 'the values of the shooting'. For that setup the observer pattern was created. The observable sends out messages about its state (here damage and interval) and the observer then displays that information. 

In your script they all happen in one and the same script, violating the Single Responsibility Principle (SRP). Both the observer pattern and the SRP are also discussed in the ebook shown on this page https://unity.com/resources/level-up-your-code-with-game-programming-patterns

Long story short: Seperate the responsibilities in the Shooting script using Actions or (Unity)Events. See links for info on how

https://github.com/watsonfilipp/DevelopingForMobileFilipp/blob/4e0c248f43efece037a656f8e3308a9b9b801506/Assets/Shooting.cs#L26
